cooked-validators/src/Cooked/Tx/Constraints.hs

```diff
@@ -100,14 +100,20 @@ instance ToLedgerConstraint OutConstraint where
           -- a different 'WithOwnStakePubKeyHash' constraint?
           <> maybe mempty Pl.ownStakePubKeyHash stak
       constr = Pl.singleton $ Pl.MustPayToPubKeyAddress (Pl.PaymentPubKeyHash p) stak mData v
-  toLedgerConstraint (PaysScript v datum value) = (lkups, constr)
+  toLedgerConstraint (PaysScript v msc datum value) = (lkups, constr)
     where
       lkups = Pl.otherScript (Pl.validatorScript v)
       constr =
-        Pl.mustPayToOtherScript
-          (Pl.validatorHash $ Pl.validatorScript v)
-          (Pl.Datum $ Pl.toBuiltinData datum)
-          value
+        Pl.singleton $
+          Pl.MustPayToOtherScript
+            (Pl.validatorHash $ Pl.validatorScript v)
+            (getStakeValidatorHash msc)
+            (Pl.Datum $ Pl.toBuiltinData datum)
+            value
+      getStakeValidatorHash :: Maybe Pl.StakingCredential -> Maybe Pl.StakeValidatorHash
```

Comment:
Is this the function you borrowed from `plutus-libs` for building a `StakeValidatorHash`? If so, please add a comment 
referring to where you found it, so we can change this in the future when things change.

Also, you're probably better off writting `getStakeValidatorHash :: Pl.StakingCredential -> Pl.StakeValidatorHash` and then use `fmap getStakeValidatorHash` to lift it to `Maybe`.

Reply:
I'll add an associated comment. But not every `StakingCredential` has an associated `StakeValidatorHash`. I can make the function `Pl.StakingCredential -> Maybe Pl.StakeValidatorHash` but then I'm not sure how to lift that correctly (likely a lack of Haskell knowledge).

Reply:
Oh, I see! Then we make it with type: `getStakeValidatorHash :: Pl.StakingCredential -> Maybe Pl.StakeValidatorHash` and we use the monad structure that `Maybe` has: `msc >>= getStakeValidatorHash`